src/app/codegen-data-model-provider/CodegenDataModelProvider_EventPathReadable.cpp

```diff
@@ -0,0 +1,169 @@
+/*
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include <app/codegen-data-model-provider/CodegenDataModelProvider.h>
+
+#include <access/AccessControl.h>
+#include <app/EventPathParams.h>
+#include <app/RequiredPrivilege.h>
+#include <app/util/af-types.h>
+#include <app/util/attribute-storage.h>
+#include <app/util/endpoint-config-api.h>
+#include <lib/core/DataModelTypes.h>
+
+namespace chip {
+namespace app {
+namespace {
+
+using namespace chip::Access;
+
+#if !CHIP_CONFIG_ENABLE_EVENTLIST_ATTRIBUTE
+bool CanAccessEvent(const Access::SubjectDescriptor & subjectDescriptor, const ConcreteClusterPath & path,
```

Comment:
So this id duplicating a bunch of code from InteractionModelEngine.cpp.  Why are we doing that?  It seems to me like this code should just live here, the "checked" mode should call into the checked data model provider, and that should compare the codegen and non-codegen provider match, or something, no?

Reply:
I expect to eventually delete the bits that are inside interactionmodelengine - those parts should have never contained direct ember calls. My understanding is that ember-compatibility-functions.cpp was the bridge in functionality, however ember calls then escaped because they were just available.

I cannot really reuse since I specifically want to separate out dependencies and depending on "codegen" is not separating things out. Because this couples "access" and ember calls, I cannot place it in ember (i.e. util) either.

For now I suggest keeping the duplication because I fully expect to delete the rest of the code later. Alternatively to deduplicate I would need to create some separate library only to pull it back later into codegen, which seems a pain.

Reply:
> since I specifically want to separate out dependencies and depending on "codegen" is not separating things out

I feel like I am missing something here.  For the "checked" bits, I had forgotten that they were done in a bespoke way instead of having an interface implementation that fans out to two other interface instances without depending on their implementations.  

But for this specific case where we are literally copy/pasting the code and there is nothing complicated about the API contract (you pass in path and subject, you get back a boolean), it seems like what we should do is basically always use the provider, and just set it to the codegen provider for now as needed, with the code moving into there.  The value of the "checked" mode seems very low to me in this case.

And I don't think we should cart around and maintain two copies of this code.  I know you plan to delete one of the copies, but there is no clear timeframe for that and in the meantime we have to maintain both copies.

Reply:
Discussed offline, will share via `#include` for now. I believe long term when/if we remove old code, these shared includes should be pulled back into codegen data provider.